modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/service/test/DDMStructureLocalServiceTest.java

```diff
@@ -92,6 +97,26 @@ public static void setUpClass() {
 		_classNameId = PortalUtil.getClassNameId(DDL_RECORD_SET_CLASS_NAME);
 	}
 
+	@Test
+	public void testAddDDMStructure() throws Exception {
+		DDMStructure ddmStructure = _addStructure(
+			RandomTestUtil.randomString());
+
+		Assert.assertTrue(
+			Validator.isNotNull(ddmStructure.getExternalReferenceCode()));
+	}
+
+	@Test(expected = DuplicateDDMStructureExternalReferenceCodeException.class)
+	public void testAddDDMStructureWithExistingExternalReferenceCode()
```

Comment:
You should cover this scenario inside the method `testAddDDMStructure()` using `AssertUtils.assertFailure`. You can see an example on class `ObjectDefinitionLocalServiceTest.java`